enumerables.rb

```diff
@@ -0,0 +1,109 @@
+module Enumerable
+
+  def my_each
+    if block_given?
+      (0..length - 1).each do |i|
+        yield(self[i])
+      end
+      self
+    else
+      Enumerable
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).each do |i|
+        yield(self[i], i)
+      end
+    else
+      Enumerable
+    end
+  end
+
+  def my_select
+    arr = []
+    if block_given?
+      (1..length - 1).each do |i|
+        arr.push(self[i]) if yield(self[i])
+      end
+    else
+      Enumerable
+    end
+    arr
+  end
+
+  def my_all?
+    results = true
+    (0..length - 1).each do |i|
+      results = yield(self[i])
+      break unless results
+    end
+    results
+  end
+
+  def my_any?
+    results = false
+    (0..length - 1).each do |i|
+      results = true if yield(self[i])
+    end
+    results
+  end
+
+  def my_none?
+    res = true
+    if block_given?
+      (0..length - 1).each do |i|
+        res = yield(self[i])
+        break
+      end
+    end
+    !res
+  end
+
+  def my_count
+    counter = 0
+    (0..length - 1).each do |i|
+      counter += 1 if yield(self[i])
+    end
+    counter
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
* [ ] enumerables my_count returns the number of items in enum through enumeration
```ruby
[1,2,3,4,5,6].my_count != [1,2,3,4,5,6].count
```

- [ ] enumerables my_count counts the number of items in enum that are equal to item if an argument is given
```ruby
[1,2,3,4,5].my_count(1) != [1,2,3,4,5].count(1)
```